django/publicmapping/redistricting/views.py

```diff
@@ -955,6 +955,7 @@ def uploadfile(request):
                 filename = '%s%s' % (dest.name, '.zip')
             else:
                 filename = dest.name
+            os.chmod(filename, 0777)
```

Comment:
It feels a little kludgey to directly change a file's permissions like this just for this one case. It'd be nice if we could get this working by setting up our permissions such that Django and Celery can both work with these shared files generated from Celery tasks.

Reply:
`NamedTemporaryFile` creates a file with mode `600` for security reasons. https://stackoverflow.com/questions/10541760/can-i-set-the-umask-for-tempfile-namedtemporaryfile-in-python

Reply:
Gotcha. Wouldn't `644` or `666` be sufficient then?

It might be good to leave a comment explaining why we need to change the permissions.